src/utils/TodoItem.tsx

```diff
@@ -0,0 +1,125 @@
+import classNames from 'classnames';
+import { FC, useState } from 'react';
+import { Todo } from '../types/Todo';
+
+interface TodoItemProps {
+  todo: Todo;
+  onDoubleClick?: () => void;
+  onClick?: (id: number) => void;
+  isLoading: boolean,
+  handleToggleComplete?: (id: number) => void;
+  isEditing?: boolean;
+  finishEditing?: () => void;
+  handleTodoTitleChange?: (todoId: number,
+    updatedData: string, currentTitle: string) => void;
+}
+
+export const TodoItem: FC<TodoItemProps> = (
+  {
+    todo,
+    onDoubleClick,
+    onClick,
+    isLoading,
+    handleToggleComplete,
+    isEditing,
+    finishEditing,
+    handleTodoTitleChange,
+  },
+) => {
+  const [editedTitle, setEditedTitle] = useState(todo?.title || '');
```

Comment:
Using todo?.title in useState may cause the state to not update when the props update. Consider moving this to useEffect to handle updates to todo.title.